sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ordering.scala

```diff
@@ -19,18 +19,28 @@ package org.apache.spark.sql.catalyst.expressions
 
 import org.apache.spark.sql.catalyst.InternalRow
 import org.apache.spark.sql.catalyst.expressions.BindReferences.bindReferences
+import org.apache.spark.sql.catalyst.expressions.codegen.GenerateOrdering
 import org.apache.spark.sql.types._
 
 
+/**
+ * A base class for generated/interpreted row ordering.
+ */
+class BaseOrdering extends Ordering[InternalRow] {
+  def compare(a: InternalRow, b: InternalRow): Int = {
```

Comment:
what if we only keep the method definition?
```
abstract class BaseOrdering extends Ordering[InternalRow] {
  def compare(a: InternalRow, b: InternalRow): Int
}
```

Reply:
Let's check in the latest commit ;)
But, my guess is that we need to implement it based on the exception message:
```
File 'generated.java', Line 5, Column 1: Non-abstract class "org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificOrdering" must implement method "public abstract int scala.math.Ordering.compare(java.lang.Object, java.lang.Object)"
	at 
...
```

Reply:
Got the same exception;
```
Error running query: org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 5, Column 1: failed to compile: org.codehaus.commons.compiler.CompileException: File 'generated.java', Line 5, Column 1: Non-abstract class "org.apache.spark.sql.catalyst.expressions.GeneratedClass$SpecificOrdering" must implement method "public abstract int scala.math.Ordering.compare(java.lang.Object, java.lang.Object)"
	at org.scalatest.Assertions.newAssertionFailedException(Assertions.scala:530)
	at org.scalatest.Assertions.newAssertionFailedException$(Assertions.scala:529)
```